File: src/nodejs/fastify/src/models/Order.js
import { z } from 'zod';

// Order model schema
export const OrderSchema = z.object({
  id: z.number(),
  user_id: z.number(),
  total_amount: z.number(),
  status: z.string(),
  created_at: z.string()
});

// OrderItem model schema
export const OrderItemSchema = z.object({
  id: z.number(),
  order_id: z.number(),
  product_name: z.string(),
  quantity: z.number(),
  price: z.number(),
  created_at: z.string()
});

// Create Order from database row
export function createOrder(row) {
  return {
    id: row.id,
    user_id: row.user_id,
    total_amount: row.total_amount,
    status: row.status,
    created_at: row.created_at
  };
}

// Create OrderItem from database row
export function createOrderItem(row) {
  return {
    id: row.id,
    order_id: row.order_id,
    product_name: row.product_name,
    quantity: row.quantity,
    price: row.price,
    created_at: row.created_at
  };
}

File: src/python/fastapi/app/models/order.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class OrderBase(BaseModel):
    """Base order model"""
    user_id: int
    total_amount: float = Field(..., gt=0)
    status: str = Field(..., min_length=1, max_length=50)


class Order(OrderBase):
    """Complete order model"""
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class OrderItemBase(BaseModel):
    """Base order item model"""
    order_id: int
    product_name: str
    quantity: int = Field(..., gt=0)
    price: float = Field(..., gt=0)


class OrderItem(OrderItemBase):
    """Complete order item model"""
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class OrderWithItems(Order):
    """Order with items"""
    items: list["OrderItem"] = []

    class Config:
        from_attributes = True
